### DISCUSSION TITLE
How do we assist?

### DISCUSSION THREAD

**Discussion by realisable** [JonPSmith/AuthPermissions.AspNetCore#116] [General]
Hello, we have successfully (although very much still in development) integrated Authpermissions into our solution with OpenIdDict.

In light of your [post](https://www.thereformedprogrammer.net/how-to-update-a-nuget-library-once-the-author-isnt-available/) we would like to support this project in any way possible.

**Comment by JonPSmith** [comment]
Hi @realisable,

Its great that you got OpenIdDict to work with AuthP. I expect others would like to have your OpenIdDict / AuthP code. At the moment I don't want to add external code to AuthP, because I'm trying to keep things simple due to to my dementia. If you can create a open-source code  /  library dealing with OpenIdDict and AuthP, then I would would be happy tweet about your work.

You might like to look at the [`IAddNewUserManager`](https://github.com/JonPSmith/AuthPermissions.AspNetCore/blob/main/AuthPermissions.SupportCode/AddUsersServices/Authentication/IAddNewUserManager.cs) where I tried an interface that could be used with various authenticate types? It might help you use the AuthP's [invite new users](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Invite-new-user-service) or [Sign up for a new tenant, with versioning](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Sign-up-for-a-new-tenant%2C-with-versioning) services. Have a look at see [this section](https://www.thereformedprogrammer.net/three-ways-to-securely-add-new-users-to-an-application-using-the-authp-library/#linking-these-two-features-to-the-asp-net-cores-authentication-handlers) of my article called "Three ways to securely add new users to an application using the AuthP library" which talks about what the [`IAddNewUserManager`](https://github.com/JonPSmith/AuthPermissions.AspNetCore/blob/main/AuthPermissions.SupportCode/AddUsersServices/Authentication/IAddNewUserManager.cs) tries to do.
